sky/provision/gcp/instance_utils.py

```diff
@@ -209,6 +247,52 @@ def delete_firewall_rule(
             firewall=firewall_rule_name,
         ).execute()
 
+    @classmethod
+    def get_vpc_name(
+        cls,
+        project_id: str,
+        zone: str,
+        instance_name: str,
+    ) -> Optional[str]:
+        # Any errors will be handled in the caller function.
+        instance = cls.load_resource().instances().get(
+            project=project_id,
+            zone=zone,
+            instance=instance_name,
+        ).execute()
+        # Format: projects/PROJECT_ID/global/networks/VPC_NAME
+        vpc_link = instance['networkInterfaces'][0]['network']
+        return vpc_link.split('/')[-1]
+
+    @classmethod
+    def create_firewall_rule(
+        cls,
+        project_id: str,
+        cluster_name_on_cloud: str,
+        port: str,
+        vpc_name: str,
+    ) -> dict:
+        name = _get_firewall_rule_name(cluster_name_on_cloud, port)
+        body = {
+            'name': name,
+            'description': f'Allow user-specified port {port} for cluster {cluster_name_on_cloud}',
+            'network': f'projects/{project_id}/global/networks/{vpc_name}',
+            'selfLink': f'projects/{project_id}/global/firewalls/' + name,
+            'direction': 'INGRESS',
+            'priority': 65534,
+            'allowed': [{
+                'IPProtocol': 'tcp',
+                'ports': [port],
```

Comment:
Does this argument support multiple arguments? Why do we create one port per rule, but in AWS we create a security group that contains all the ports?

Reply:
Now it will only create one rule 🫡